File: GeneratedEntityReaderClasses/Components/KickComponent.cs
namespace GeneratedNoitaClasses;
public class KickComponent : NoitaComponentBase,  iNoitaType<KickComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool can_kick { get; set; } = new();
	public NoitaLensValue<NoitaFloat> max_force { get; set; } = new();
	public NoitaLensValue<NoitaFloat> player_kickforce { get; set; } = new();
	public NoitaFloat kick_radius { get; set; } = new();
	public NoitaLensValue<NoitaFloat> kick_damage { get; set; } = new();
	public NoitaLensValue<NoitaFloat> kick_knockback { get; set; } = new();
	public NoitaFloat telekinesis_throw_speed { get; set; } = new();
	public NoitaString kick_entities { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		can_kick.Read(s);
		max_force.Read(s);
		player_kickforce.Read(s);
		kick_radius.Read(s);
		kick_damage.Read(s);
		kick_knockback.Read(s);
		telekinesis_throw_speed.Read(s);
		kick_entities.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		can_kick.Write(s);
		max_force.Write(s);
		player_kickforce.Write(s);
		kick_radius.Write(s);
		kick_damage.Write(s);
		kick_knockback.Write(s);
		telekinesis_throw_speed.Write(s);
		kick_entities.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/KickComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaLensValue
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class KickComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    can_kick:NoitaBool = field(default_factory=NoitaBool)
    max_force:NoitaLensValue = field(default_factory=lambda: NoitaLensValue(NoitaFloat))
    player_kickforce:NoitaLensValue = field(default_factory=lambda: NoitaLensValue(NoitaFloat))
    kick_radius:NoitaFloat = field(default_factory=NoitaFloat)
    kick_damage:NoitaLensValue = field(default_factory=lambda: NoitaLensValue(NoitaFloat))
    kick_knockback:NoitaLensValue = field(default_factory=lambda: NoitaLensValue(NoitaFloat))
    telekinesis_throw_speed:NoitaFloat = field(default_factory=NoitaFloat)
    kick_entities:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.can_kick.Read(s)
        self.max_force.Read(s)
        self.player_kickforce.Read(s)
        self.kick_radius.Read(s)
        self.kick_damage.Read(s)
        self.kick_knockback.Read(s)
        self.telekinesis_throw_speed.Read(s)
        self.kick_entities.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.can_kick.Write(s)
        self.max_force.Write(s)
        self.player_kickforce.Write(s)
        self.kick_radius.Write(s)
        self.kick_damage.Write(s)
        self.kick_knockback.Write(s)
        self.telekinesis_throw_speed.Write(s)
        self.kick_entities.Write(s)

